packages/relay/src/lib/errors/MirrorNodeClientError.ts

```diff
@@ -21,30 +21,49 @@
 
 export class MirrorNodeClientError extends Error {
     public statusCode: number;
+    public errorMessage?: string;
 
     static retryErrorCodes: Array<number> = [400, 404, 408, 425, 500]
 
     static ErrorCodes = {
       ECONNABORTED: 504,
-      CONTRACT_REVERT_EXECUTED : 400
+      CONTRACT_REVERT_EXECUTED : 400,
+      NOT_SUPPORTED: 501
     };
 
     static statusCodes = {
       NOT_FOUND: 404
     };
 
-    constructor(message: string, statusCode: number) {
-      super(message);
-      this.statusCode = statusCode;
+    constructor(error: any, statusCode: number) {
+        if (error.response?.data?._status?.messages?.length) {
+            const msg = error.response.data._status.messages[0];
+            const {message, data} = msg;
+            super(message);
 
-      Object.setPrototypeOf(this, MirrorNodeClientError.prototype);
+            this.errorMessage = data;
+        }
+        else {
+            super(error.message);
+        }
+
+        this.statusCode = statusCode;
+        Object.setPrototypeOf(this, MirrorNodeClientError.prototype);
     }
 
     public isTimeout(): boolean {
       return this.statusCode === MirrorNodeClientError.ErrorCodes.ECONNABORTED;
     }
 
+    public isContractReverted(): boolean {
+        return this.statusCode === MirrorNodeClientError.ErrorCodes.CONTRACT_REVERT_EXECUTED;
+    }
+
     public isNotFound(): boolean {
       return this.statusCode === MirrorNodeClientError.statusCodes.NOT_FOUND;
     }
+
+    public isNotSupported(): boolean {
```

Comment:
Q: Is it okay to have an and here or is the name incorrect?
Are we aware of any unsupported cases that aren't precompile?
Should this instead be `isUnsupportedSystemContractOperaton()` with a separate `isNotSupported()`?

Reply:
Hm, yes, i've added a separate error for this.

Reply:
Gotcha, i thought maybe they were connected but this works for now